fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +318,24 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replace Lucene Index with a new index.
+     *
+     * @param newLuceneIndexPath  The path of the new index
+     */
+    public void replaceIndex(String newLuceneIndexPath) {
+        luceneIndexPath = newLuceneIndexPath;
+
```

Comment:
Make sense. I will

1. lock and close the old dir
2. move the old dir to a different location
3. move the new dir to the old location
4. unlock the old dir and file system delete it